qt/aqt/webview.py

```diff
@@ -654,6 +660,14 @@ def _onBridgeCmd(self, cmd: str) -> Any:
             self._maybeRunActions()
         elif cmd == "close":
             self.onEsc()
+        elif cmd.startswith("op_executed"):
+            mw.autosave()
+            changes = OpChanges()
+            Parse(cmd[len("op_executed") + 1 :], changes)
+            gui_hooks.operation_did_execute(changes, self.parent())
+            if mw.col.op_made_changes(changes):
+                gui_hooks.state_did_reset()
```

Comment:
Will give this a proper review tomorrow - just a brief comment for now. A bridge command shouldn't be required here -  instead, you should be getting the actual changes from the output of the importCsv/importAnkiPackage command. mediasrv.py calls the _raw version which takes and returns bytes; you could remove these from exposed_backend_list and instead implement intercept routines manually for them (putting them in post_handler_list). The routes could call the _raw method, get the raw return bytes, and before returning the bytes to the web frontend, it could parse them as an ImportResponse and feed them into the undo infrastructure. Rather than repeating the contents of  _finish_op, that code should probably be split into a free function so it can be called by both CollectionOp and by code like this.

Bear in mind that mediasrv runs in the background, so the GUI code should be called with run_on_main, like mediasrv's update_deck_configs